### PR TITLE
UCT/IB: Move few slow-path functions to C file

### PR DESCRIPTION
**Pull request by yosefe**
# Why
Code cleanup and preparation for multi-path (RoCE LAG/LMC) support in UCT

### CODE REVIEW

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -435,6 +435,44 @@ ucs_status_t uct_ib_iface_create_ah(uct_ib_iface_t *iface,
...
+
+void uct_ib_iface_fill_ah_attr_from_addr(uct_ib_iface_t *iface,
+                                         const uct_ib_address_t *ib_addr,
+                                         struct ibv_ah_attr *ah_attr)
+{
+    union ibv_gid  gid;
+    uint16_t       lid;
+
+    ucs_assert(!uct_ib_iface_is_roce(iface) ==
+               !(ib_addr->flags & UCT_IB_ADDRESS_FLAG_LINK_LAYER_ETH));
```
Comment (dmitrygx) [inline src/uct/ib/base/ib_iface.c]: minor: Can we remove `!` for left and right values in the statement to simplify logic?

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 17-Feb-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/dfe57032aa29df1e74a008648aaf01d12e0c96d5/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/88ef31074496359b45569bdc0337d8bd3abc8af3/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/a246b91143b0c309df36beafb452e566cb449cbf/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/854b6e03374e05ad3a207a50df53e1c0dcca5464/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/f5c759c0daf874cb85bb63fdc0c935de05443d1e/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/ed577449a007f8348dd2295189b9a2c47c9ffe13/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/7370b251c9c89b4ef49be2a8ead52b865ca43956/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/bf8acc42be1253622879085b1630409404af86fc/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/b8ebf904a91a40b0506a273bcc2ee0891059b6bd/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/3ff900d98b644c716398b121db0fa9b84592c148/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/7ae334aa7a85d950dfef9e4457d596dca2492f16/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/e0bf0be3770ad4f9185e973d2273e455bd6e1d98/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/2ad31e030cb1b0ce28277738b12c4f8621f79035/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/86f357fc72a81916890ee2943146f75566059894/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/d7f5f143355434ea96e660e6f4604772e4ad92e5/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/ae30161359bd87cd6d59ab52b57b14de324ed993/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/04abeff4f117e51a2f14b14ac562f3e1064558d7/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/4eec86e8fb27e07c1de69ae86b31243323440db8/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/31cdaa995dcf92b31e33e6ee9e6a7497d1cdabff/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/41c1ba8d8bfd63f1c64ca1d7ba260d8fdf797232/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/ce48bcb00e696b68aba6ad1f7d3d39993a12bca9/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/891641f26f857eede413b9a0ea3b9996e1202cd2/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/8f4a2e8a3cac50457918724012c5c1d3dfb0ea57/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/97f5b903156d48553eb540254cd3b9b9aa72d9df/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/bb6c4141e9727c03af706ca9836c852e/raw/9a0c0c83dfb9bf1c451baf294c20e0b2c921d570/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: src/uct/ib/base/ib_iface.c (hunk 1 above)
Comment (yosefe) [reply src/uct/ib/base/ib_iface.c]: unfortunately not, (ib_addr->flags & UCT_IB_ADDRESS_FLAG_LINK_LAYER_ETH) may not be equal to 1

Comment (dmitrygx) [reply src/uct/ib/base/ib_iface.c]: Ah, I see

**Pull request merged**